test/common.py

```diff
@@ -0,0 +1,159 @@
+# use this file except in compliance with the License. A copy of the License
+# is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import os
+import random
+import sys
+from tempfile import TemporaryDirectory
+from typing import Optional, Tuple
+from unittest.mock import patch
+
+import mxnet as mx
+import numpy as np
+
+import sockeye.bleu
+import sockeye.constants as C
+import sockeye.train
+import sockeye.translate
+import sockeye.utils
+
+
+def gaussian_vector(shape, return_symbol=False):
+    """
+    Generates random normal tensors (diagonal covariance)
+
+    :param shape: shape of the tensor.
+    :param return_symbol: True if the result should be a Symbol, False if it should be an Numpy array.
+    :return: A gaussian tensor.
+    """
+    return mx.sym.random_normal(shape=shape) if return_symbol else np.random.normal(size=shape)
+
+
+def integer_vector(shape, max_value, return_symbol=False):
+    """
+    Generates a random positive integer tensor
+
+    :param shape: shape of the tensor.
+    :param max_value: maximum integer value.
+    :param return_symbol: True if the result should be a Symbol, False if it should be an Numpy array.
+    :return: A random integer tensor.
+    """
+    return mx.sym.round(mx.sym.random_uniform(shape=shape) * max_value) if return_symbol \
+        else np.round(np.random.uniform(size=shape) * max_value)
+
+
+def uniform_vector(shape, min_value=0, max_value=1, return_symbol=False):
+    """
+    Generates a uniformly random tensor
+
+    :param shape: shape of the tensor
+    :param min_value: minimum possible value
+    :param max_value: maximum possible value (exclusive)
+    :param return_symbol: True if the result should be a mx.sym.Symbol, False if it should be a Numpy array
+    :return:
+    """
+    return mx.sym.random_uniform(low=min_value, high=max_value, shape=shape) if return_symbol \
+        else np.random.uniform(low=min_value, high=max_value, size=shape)
+
+
+def generate_random_sentence(vocab_size, max_len):
+    """
+    Generates a random "sentence" as a list of integers.
+
+    :param vocab_size: Number of words in the "vocabulary". Note that due to
+                       the inclusion of special words (BOS, EOS, UNK) this does *not*
+                       correspond to the maximum possible value.
+    :param max_len: maximum sentence length.
+    """
+    length = random.randint(1, max_len)
+    # Due to the special words, the actual words start at index 3 and go up to vocab_size+2
+    return [random.randint(3, vocab_size + 2) for _ in range(length)]
+
+
+_DIGITS = "0123456789"
+
+
+def generate_digits_file(source_path: str,
```

Comment:
this method and also run_train_translate seem to be only relevant to the system tests, so I'd consider moving it in the system module.